Packet++/header/StpLayer.h

```diff
@@ -0,0 +1,834 @@
+	};
+#pragma pack(pop)
+
+/**
+ * @struct rstp_conf_bpdu
+ * Represents payload configuration of BPDU for Rapid STP (RSTP)
+ */
+#pragma pack(push, 1)
+	struct rstp_conf_bpdu
+	{
+		/// Protocol ID. Fixed at 0x0, which represents IEEE 802.1d
+		uint16_t protoId;
+		/// Protocol version. 0x2 for RSTP
+		uint8_t version;
+		/// Type of the BPDU. 0x2 for RSTP/MSTP
+		uint8_t type;
+		/// Flag for indicate purpose of BPDU
+		uint8_t flag;
+		/// Root bridge ID
+		uint64_t rootId;
+		/// Cost of path
+		uint32_t pathCost;
+		/// Bridge ID
+		uint64_t bridgeId;
+		/// Port ID
+		uint16_t portId;
+		/// Age of the BPDU
+		uint16_t msgAge;
+		/// Maximum age of the BPDU
+		uint16_t maxAge;
+		/// BPDU transmission interval
+		uint16_t helloTime;
+		/// Delay for STP
+		uint16_t forwardDelay;
+		/// Version1 length. The value is 0x0
+		uint8_t version1Len;
+	};
+#pragma pack(pop)
+
+/**
+ * @struct mstp_conf_bpdu
+ * Represents payload configuration of BPDU for Multiple STP (MSTP)
+ */
+#pragma pack(push, 1)
+	struct mstp_conf_bpdu
+	{
+		/// Protocol ID. Fixed at 0x0, which represents IEEE 802.1d
+		uint16_t protoId;
+		/// Protocol version. 0x3 for MSTP
+		uint8_t version;
+		/// Type of the BPDU. 0x2 for RSTP/MSTP
+		uint8_t type;
+		/// Flag for indicate purpose of BPDU
+		uint8_t flag;
+		/// Root bridge ID
+		uint64_t rootId;
+		/// Cost of path
+		uint32_t pathCost;
+		/// Bridge ID
+		uint64_t bridgeId;
+		/// Port ID
+		uint16_t portId;
+		/// Age of the BPDU
+		uint16_t msgAge;
+		/// Maximum age of the BPDU
+		uint16_t maxAge;
+		/// BPDU transmission interval
+		uint16_t helloTime;
+		/// Delay for STP
+		uint16_t forwardDelay;
+		/// Version1 length. The value is 0x0
+		uint8_t version1Len;
+		/// Version3 length.
+		uint16_t version3Len;
+		/// Configuration id format selector
+		uint8_t mstConfigFormatSelector;
+		/// Configuration id name
+		uint8_t mstConfigName[32];
+		/// Configuration id revision
+		uint16_t mstConfigRevision;
+		/// Configuration id digest
+		uint8_t mstConfigDigest[16];
+		/// CIST internal root path cost
+		uint32_t irpc;
+		/// CIST bridge id
+		uint64_t cistBridgeId;
+		/// CIST remaining hop count
+		uint8_t remainId;
+	};
+#pragma pack(pop)
+
+/**
+ * @struct msti_conf_msg
+ * Represents MSTI configuration messages. Each message contains 16 bytes and MSTP can contain 0 to 64 MSTI messages.
+ */
+#pragma pack(push, 1)
+	struct msti_conf_msg
+	{
+		/// MSTI flags
+		uint8_t flags;
+		/// Regional root switching id (Priority (4 bits) + ID (12 bits) + Regional root (48 bits - MAC address))
+		uint64_t regionalRootId;
+		/// Total path cost from local port to regional port
+		uint32_t pathCost;
+		/// Priority value of switching device
+		uint8_t bridgePriority;
+		/// Priority value of port
+		uint8_t portPriority;
+		/// Remaining hops of BPDU
+		uint8_t remainingHops;
+	};
+#pragma pack(pop)
+
+	/**
+	 * @class StpLayer
+	 * Represents an Spanning Tree Protocol Layer
+	 */
+	class StpLayer : public Layer
```

Comment:
I think we can add crafting/editing capabilities to the STP layer classes quite easily. I guess you didn't do it because you didn't have time?
I think it's fine for now. If someone asks we can add it pretty quickly

Reply:
@seladb Actually I have same problem here with divide 256. I can't find enough document or information (at least in documents that I can understand) which clearly states what will happen if a struct field changes. What I mean if the user changed some field, is there a need for `calculateComputeFields` function? I can agree this won't be a requirement but if there is a need I think it should be there.

When I made a research again about the protocol and if there is no dependency between the fields I agree we can add crafting/editing support later on a separate PR easily

Reply:
That's a good point! Usually we don't care about message validity, meaning we let users set any field to any (legal) value and we don't care if the message is valid or not. This removes a lot of complexity and let users craft invalid messages for testing purposes etc. 
We do care about field types - meaning we don't let a user set an int in a string field, or if a field is an enum its corresponding `set` method will accept the enum as a parameter.

Please let me know if it makes sense